woltdater/utils/handlers.py

```diff
@@ -18,18 +29,48 @@ async def check_status(message: types.Message, memory_plugin: AbstractMemoryPlug
     Gets message and subscribe client if needed
     """
     logging.info(f'Trying to get venue status for {message.text}')
-    restaurant_symbol = get_restaurant_symbol_from_url(message.text.lower())
-    try:
-        venue_status = await is_venue_available_status(restaurant_symbol)
-        if venue_status:
-            await message.reply(VENUE_OPEN_MSG.format(restaurant_symbol=restaurant_symbol))
+
+    if "https://" in message.text:
+        restaurant_symbol = get_restaurant_symbol_from_url(message.text.lower())
+        try:
+            venue_status = await is_venue_available_status(restaurant_symbol)
+            if venue_status:
+                await message.reply(VENUE_OPEN_MSG.format(restaurant_symbol=restaurant_symbol))
+            else:
+                await message.reply(WILL_UPDATE_MSG.format(restaurant_symbol=restaurant_symbol))
+                await memory_plugin.subscribe(restaurant_symbol, message.chat.id)
+        except RestaurantWasNotFoundException:
+            await message.reply(RESTAURANT_NOT_FOUND_MSG)
+    else:
+        venue = await search_vanue_by_name(message.text)
+
+        if venue:
+            if len(venue) > 1:
+                kb = keyboard_maker(venue)
+                await message.answer(SEARCH_LIST_TITLE, reply_markup=kb)
+            else:
+                if venue[0]["online"]:
+                    await message.answer(VENUE_OPEN_MSG.format(name=venue[0]['name']))
```

Comment:
You are right 😅  
Changed the MSG to include venue name but removed the MSG form the PR and forgot about this format 🤦‍♂️
I still think it is nicer to reply with the venue name instead of the slug... 
I implemented it (in my repo) by querying the wolt API every time we want to reply with the venue name, but a better solution might be to save an the name with the restaurant symbol, but that calls for a bigger change. Tell me what you think.